### PR TITLE
chore: move some components from demosplan-ui to Core and rename it.

### PR DESCRIPTION
**Pull request by ahmad-demos**
https://demosdeutschland.slack.com/archives/C5BA8ABB2/p1697635616944349?thread_ts=1697526096.824259&cid=C5BA8ABB2

- move some components from demosplan-ui to Core and rename it.
- DpTextWrapper -> rename to TextContentRenderer
- DpHeightLimit -> rename to HeightLimit
- DpSwitcher

### CODE REVIEW

File: client/js/components/shared/TextContentRenderer.vue [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<script>
+import { DpLoading } from '@demos-europe/demosplan-ui'
+
+export default {
+  name: 'DpTextWrapper',
```
Comment (spiess-demos) [inline client/js/components/shared/TextContentRenderer.vue:5]: The component name should match the file name.

File: client/js/components/statement/assessmentTable/DpFragmentsSwitcher.vue [hunk 2]
```diff
@@ -54,7 +54,7 @@ statements and fragments is now in DpAssessmentTableCard.vue
 
 <script>
 import { mapGetters, mapState } from 'vuex'
-import { DpSwitcher } from '@demos-europe/demosplan-ui'
+import { DpSwitcher } from './DpSwitcher'
```
Comment (spiess-demos) [inline client/js/components/statement/assessmentTable/DpFragmentsSwitcher.vue:57]: Actually i meant to move it _inside_ DpFragmentsSwitcher.vue.

**Review by spiess-demos — CHANGES_REQUESTED** [review]

File: client/js/components/shared/TextContentRenderer.vue (hunk 1 above)
Comment (ahmad-demos) [reply client/js/components/shared/TextContentRenderer.vue]: Done in https://github.com/demos-europe/demosplan-core/pull/2169/commits/98d20583e3012ef3e7021f0eb95fdea06ae459ca

File: client/js/components/statement/assessmentTable/DpFragmentsSwitcher.vue (hunk 2 above)
Comment (ahmad-demos) [reply client/js/components/statement/assessmentTable/DpFragmentsSwitcher.vue]: Done in https://github.com/demos-europe/demosplan-core/pull/2169/commits/98d20583e3012ef3e7021f0eb95fdea06ae459ca

File: client/js/components/statement/assessmentTable/DpFragmentsSwitcher.vue [hunk 3]
```diff
@@ -168,6 +170,10 @@ export default {
     showAllFragments () {
       this.allFragmentsShown = !this.allFragmentsShown
       this.$emit('fragments:showall', this.allFragmentsShown)
+    },
+
+    toggle () {
+      this.$emit('input', (this.statementTabVisible === false))
```
Comment (spiess-demos) [inline client/js/components/statement/assessmentTable/DpFragmentsSwitcher.vue:176]: This has to be changed to

```suggestion
      this.$emit('toggletabs', (this.statementTabVisible === false))
```

**Review by spiess-demos — CHANGES_REQUESTED** [review]

**Review by spiess-demos — CHANGES_REQUESTED** [review]
Last nitpick: In DemosPlanCoreBundle/Controller/AssessmentTable/DemosPlanAssessmentTableController.php:816 there is a `<dp-height-limit>` left.

**Pull request merged**